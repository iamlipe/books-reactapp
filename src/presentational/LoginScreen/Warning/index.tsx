import React, {memo} from 'react';
import styled from 'styled-components/native';

import WarningBox from '@assets/svgs/warning-box.svg';

interface Props {
  error: string;
}

const Warning = ({error}: Props) => {
  return (
    <StyledWarningContainer>
      <StyledWarningBox />
      <StyledText>{error}</StyledText>
    </StyledWarningContainer>
  );
};

const StyledWarningContainer = styled.View`
  width: 70%;
  max-width: 239px;
  height: 48px;

  align-items: center;
  justify-content: center;

  padding: 12px 8px 8px 8px;
`;

const StyledWarningBox = styled(WarningBox)`
  position: absolute;
`;

const StyledText = styled.Text`
  color: ${({theme}) => theme.colors.TEXT_LIGHT};
`;

export default memo(Warning);
