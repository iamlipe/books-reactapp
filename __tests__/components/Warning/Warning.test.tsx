import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@config/functions/Testing';

import Warning from '@presentational/LoginScreen/Warning';

describe('Warning', () => {
  const error = 'something went wrong';

  test('should render correctly', () => {
    renderWithThemeProvider(<Warning error={error} />);
  });

  test('should match snapshot', () => {
    matchSnapshotWithProvider(<Warning error={error} />);
  });

  test('should render error', () => {
    const warning = renderWithThemeProvider(<Warning error={error} />);

    const errorWarning = warning.queryByText(error);

    expect(errorWarning).toBeTruthy();
  });
});
