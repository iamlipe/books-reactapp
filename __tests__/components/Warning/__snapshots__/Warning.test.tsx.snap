// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warning should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 48,
        "justifyContent": "center",
        "maxWidth": 239,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
        "width": "70%",
      },
    ]
  }
>
  <SvgMock
    style={
      Array [
        Object {
          "position": "absolute",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    something went wrong
  </Text>
</View>
`;
